File: Android App/app/src/main/java/revert/shredsheets/models/scales/MajorScale.java
package revert.shredsheets.models.scales;

public class MajorScale extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{2, 2, 1, 2, 2, 2, 1};
    }

    @Override
    public String getName() {
        return "Major";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{"Ionian (Major)", "Dorian", "Phrygian", "Lydian", "Mixolydian", "Aeolian (Minor)", "Locrian"};
    }

    @Override
    protected boolean[][] getModalHighlighting() {
        return new boolean[][]{
                {true, false, true, false, false, true, true},
                {true, false, true, false, false, true, true},
                {true, true, true, false, false, true, true},
                {true, false, true, true, true, false, false},
                {true, false, true, false, false, false, true},
                {true, false, true, false, false, true, true},
                {true, true, true, false, true, false, true}
        };
    }

    @Override
    public int[][] getDegrees() {
        return new int[][]{
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7}
        };
    }

    @Override
    public String[] getChords() {
        return new String[]{"M7", "m7", "m7", "M7", "7", "m7", "m7ø"};
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/MajorScale.kt
package com.bm.shredsheets.models.scales

class MajorScale : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(2, 2, 1, 2, 2, 2, 1)
    override val name: String
        get() = "Major"


    override fun getModeNames(): Array<String> {
        return arrayOf("Ionian (Major)", "Dorian", "Phrygian", "Lydian", "Mixolydian", "Aeolian (Minor)", "Locrian")
    }

    override val getEnum: Scales
        get() = Scales.Major

    override val modalHighlighting: Array<BooleanArray>
        protected get() = arrayOf(
            booleanArrayOf(true, false, true, false, false, true, true),
            booleanArrayOf(true, false, true, false, false, true, true),
            booleanArrayOf(true, true, true, false, false, true, true),
            booleanArrayOf(true, false, true, true, true, false, false),
            booleanArrayOf(true, false, true, false, false, false, true),
            booleanArrayOf(true, false, true, false, false, true, true),
            booleanArrayOf(true, true, true, false, true, false, true)
        )

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7)
        )

    override val chords: Array<String>
        get() = arrayOf("M7", "m7", "m7", "M7", "7", "m7", "m7ø")
}

